src/middleware/authentication/middleware.ts

```diff
@@ -1,18 +1,37 @@
+// biome-ignore lint/correctness/noNodejsModules: we need the node crypto module
+import { createHash } from "node:crypto";
 import type { NextFunction, Request, Response } from "express";
 import { introspectToken } from "@lib/idp/introspectToken";
+import { RedisConnector } from "@src/lib/redisConnector";
+
+const cacheExpiry = 300; // seconds
+
+export function getAccessTokenCacheKey(accessToken: string): string {
```

Comment:
I believe the `export` is not needed here.

```suggestion
function getAccessTokenCacheKey(accessToken: string): string {
```

Reply:
Heh, this is a case of me not having properly finished my unit testing coverage yet 😅  I'm planning to add unit tests for this function directly (hopefully later today or tomorrow).

Reply:
That is also what I was thinking ;) It is usually an indicator of a function that should live elsewhere.

After reviewing the whole middleware function, I would be in favor of having the caching layer functions implemented in a library file (probably under the `idp` folder). Even though it is not much code in this case, it would feel cleaner.

FYI: Since I experience it once, Biome has a check for functions that are "too complex" because of too many lines of code or too much conditional logic.

Reply:
Ah gotcha, I like that idea of moving the caching layer into a library file to keep things clean.  

That's also interesting to hear about Biome's complexity check - I can see that being super handy as things keep progressing on the API.

Reply:
I've moved the introspection cache logic into its own library file.  Let me know if this is what you were thinking!  If yes, I'll get the unit tests updated.

Reply:
Nice. This is what I was thinking ;) I see the `getAccessTokenCacheKey` function is still exported for testing I suppose. Since it is more of an internal function you could hide it and it will get tested through the other two functions.

Reply:
Sounds good, I'll remove the `export` and test through the other two functions.